internal/impl/kube.go

```diff
@@ -173,9 +174,30 @@ type KubeConfig struct {
 	Observability map[string]string
 }
 
+// shortenComponent shortens the given component name to be of the format
+// <pkg>-<IfaceType>. (Recall that the full component name is of the format
+// <path1>/<path2>/.../<pathN>/<IfaceType>.)
+func shortenComponent(component string) string {
```

Comment:
can we use the weaver/runtime/logging/pretty.go#ShortenComponent? Or move that one somewhere else to reuse it?

Reply:
`logging.ShortenComponent` returns `collatz.Odd`, while `shortenComponent` returns `collatz-Odd`. If you prefer, I can call `logging.ShortenComponent` and then call `strings.ReplaceAll` to replace `/` with `-`?